daemon/src/model/cfd.rs

```diff
@@ -700,14 +706,59 @@ pub enum CfdStateChangeEvent {
     CommitTxSent,
 }
 
+/// Returns the Profit/Loss (P/L) as Bitcoin. Losses are capped by the provided margin
 fn calculate_profit(
-    _intial_price: Usd,
-    _current_price: Usd,
-    _interest_per_day: Decimal,
-    _fee: Usd,
-) -> Result<(Amount, Usd)> {
-    // TODO: profit calculation
-    Ok((Amount::ZERO, Usd::ZERO))
+    initial_price: Usd,
+    current_price: Usd,
+    quantity: Usd,
+    margin: Amount,
+) -> Result<(Decimal, Decimal)> {
+    let margin = Decimal::from_f64(margin.as_btc()).context("Expect to be a valid decimal")?;
```

Comment:
I would still do the conversion from the `u64` sat and avoid float here. You can always shift the Decimal.
I see that this would result in the same thing, but I would keep it consistent in not using `from_f64` at all if possible.

Reply:
I opted for using Bitcoin for the UI and hence I use `as_btc` here and avoid back and forth conversion.

Reply:
Please read this: https://stackoverflow.com/a/3730040/2489334

Using float for monetary calculations ALWAYS includes loss of precision.

There is no "back and forth here". `Amount`s are already in sats internally, the only conversation to BTC should happen in the UI / API layer.

Reply:
With back and fourth I mean something what we do in `margin()`: 

```
    let sat_adjust = Decimal::from(Amount::ONE_BTC.as_sat()).into();
    let margin = margin.checked_mul(sat_adjust)?;
    let margin = Amount::from_sat(margin.try_into_u64()?);
```

I'm happy to use `from_u64` instead.

Reply:
Actually I change my mind. Imho in this case it does not make much sense to use sats: `profit_btc` is in _Bitcoin_ (resulting from the formula we use) and I only need the `margin` for subtracting it from the `profit_btc`. 

The other time where `margin` is used is for calculating the `profit_in_percent` which is not a crucial value and should be rounded anyways. 

Any more thoughts on that?

Reply:
I still think we should do the "back-and-forth" because every time I would look at this code I'd have to think about if the `f64` conversion should not be removed here. I would change it for consistency reasons, even if it results in more "unnecessary" adjustment steps for this particular instance.

Reply:
No worries, I'll do it ;)